code/modules/basketball/hoop.dm

```diff
@@ -0,0 +1,195 @@
+
+/obj/structure/hoop/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/simple_rotation, ROTATION_REQUIRE_WRENCH|ROTATION_IGNORE_ANCHORED, AfterRotation = CALLBACK(src, PROC_REF(reset_appearance)))
+	update_appearance()
+	register_context()
+
+/obj/structure/hoop/add_context(atom/source, list/context, obj/item/held_item, mob/living/user)
+	context[SCREENTIP_CONTEXT_CTRL_LMB] = "Reset score"
+	return CONTEXTUAL_SCREENTIP_SET
+
+/obj/structure/hoop/proc/reset_appearance()
+	update_appearance()
+
+/obj/structure/hoop/proc/score(obj/item/toy/basketball/ball, mob/living/baller, points)
+	// we still play buzzer sound regardless of the object
+	playsound(src, 'sound/machines/scanbuzz.ogg', 100, FALSE)
+
+	if(!istype(ball))
+		return
+
+	total_score += points
+	update_appearance()
+	// whoever scored doesn't get to pickup the ball instantly
+	COOLDOWN_START(ball, pickup_cooldown, PICKUP_RESTRICTION_TIME)
+
+	ball.pickup_restriction_ckeys |= baller.ckey
+	return TRUE
+
+/obj/structure/hoop/update_overlays()
+	. = ..()
+
+	if(dir & NORTH)
+		SET_PLANE_IMPLICIT(src, GAME_PLANE_UPPER)
+
+	cut_overlays()
+
+	var/dir_offset_x = 0
+	var/dir_offset_y = 0
+
+	switch(dir)
+		if(NORTH)
+			dir_offset_y = -32
+		if(SOUTH)
+			dir_offset_y = 32
+		if(EAST)
+			dir_offset_x = -32
+		if(WEST)
+			dir_offset_x = 32
+
+	var/mutable_appearance/scoreboard = mutable_appearance('icons/obj/signs.dmi', "basketball_scorecard")
+	scoreboard.pixel_x = dir_offset_x
+	scoreboard.pixel_y = dir_offset_y
+	SET_PLANE_EXPLICIT(scoreboard, GAME_PLANE, src)
+	. += scoreboard
+
+	var/ones = total_score % 10
+	var/mutable_appearance/ones_overlay = mutable_appearance('icons/obj/signs.dmi', "days_[ones]", layer + 0.01)
+	ones_overlay.pixel_x = 4
+	var/mutable_appearance/emissive_ones_overlay  = emissive_appearance('icons/obj/signs.dmi', "days_[ones]", src, alpha = src.alpha)
+	emissive_ones_overlay.pixel_x = 4
+	scoreboard.add_overlay(ones_overlay)
+	scoreboard.add_overlay(emissive_ones_overlay)
+
+	var/tens = (total_score / 10) % 10
+	var/mutable_appearance/tens_overlay = mutable_appearance('icons/obj/signs.dmi', "days_[tens]", layer + 0.01)
+	tens_overlay.pixel_x = -5
+
+	var/mutable_appearance/emissive_tens_overlay  = emissive_appearance('icons/obj/signs.dmi', "days_[tens]", src, alpha = src.alpha)
+	emissive_tens_overlay.pixel_x = -5
+	scoreboard.add_overlay(tens_overlay)
+	scoreboard.add_overlay(emissive_tens_overlay)
+
+/obj/structure/hoop/attackby(obj/item/ball, mob/living/baller, params)
+	if(get_dist(src, baller) < 2) // TK users aren't allowed to dunk
+		if(baller.transferItemToLoc(ball, drop_location()))
+			var/dunk_dir = get_dir(baller, src)
+
+			var/dunk_pixel_y = dunk_dir & SOUTH ? -16 : 16
+			var/dunk_pixel_x = dunk_dir & EAST && 16 || dunk_dir & WEST && -16 || 0
+
+			animate(baller, pixel_x = dunk_pixel_x, pixel_y = dunk_pixel_y, time = 5, easing = BOUNCE_EASING|EASE_IN|EASE_OUT)
+			sleep(0.5 SECONDS)
+			animate(baller, pixel_x = 0, pixel_y = 0, time = 3)
```

Comment:
Would probably be better to queue the animate up with a timer as opposed to stringing it together with a sleep, since attackby could get spammed rather easily (Or ASYNC it, just confirm that the animations will flow together so that it has the right effect).

Reply:
Applied with 38c57cf0b8d951fc3c96d1dbe53f0baf637b39ca but I need to do some testing